source/acquaman/AMContinuousScanActionController.cpp

```diff
@@ -0,0 +1,513 @@
+		baseAmptekTimeScale = AMBeamline::bl()->exposedDetectorByName("AmptekSDD1")->amdsPollingBaseTimeMilliseconds();
+	if(requiredBufferNames.contains("Scaler (BL1611-ID-1)"))
+		baseScalerTimeScale = AMBeamline::bl()->exposedDetectorByName("TEY")->amdsPollingBaseTimeMilliseconds();
+
+	int largestBaseTimeScale;
+	if(baseScalerTimeScale < baseAmptekTimeScale)
+		largestBaseTimeScale = baseAmptekTimeScale;
+	else
+		largestBaseTimeScale = baseScalerTimeScale;
+
+
+	AMDSClientDataRequest *scalerClientDataRequst = clientDataRequestMap_.value("Scaler (BL1611-ID-1)");
+	int totalCount = scalerClientDataRequst->data().count();
+	int rebasedTotalCount = (totalCount*baseScalerTimeScale)/largestBaseTimeScale;
+	qDebug() << "Original totalCount " << totalCount << " rebasedTotalCount " << rebasedTotalCount;
+
+	QMap<QString, QVector<qint32> > scalerChannelVectors;
+	QMap<QString, qint32> scalerChannelRunningSums;
+	QMap<int, QString> scalerChannelIndexMap;
+	CLSScalerChannelDetector *asScalerChannelDetector = 0;
+	for(int x = 0, size = generalConfig_->detectorConfigurations().count(); x < size; x++){
+		AMDetector *oneDetector = AMBeamline::bl()->exposedDetectorByInfo(generalConfig_->detectorConfigurations().at(x));
+		asScalerChannelDetector = qobject_cast<CLSScalerChannelDetector*>(oneDetector);
+		if(asScalerChannelDetector){
+			scalerChannelIndexMap.insert(asScalerChannelDetector->enabledChannelIndex(), oneDetector->name());
+			scalerChannelVectors.insert(oneDetector->name(), QVector<qint32>(rebasedTotalCount, 0));
+			scalerChannelRunningSums.insert(oneDetector->name(), 0);
+		}
+	}
+
+	// I think we can do a check against the number of actually enabled channels on the scaler ... just can't figure out a good way to get that value right now
+//	AMDSLightWeightScalarDataHolder *asScalarDataHolder = qobject_cast<AMDSLightWeightScalarDataHolder*>(scalerClientDataRequst->data().at(0));
+//	if(asScalarDataHolder && asScalarDataHolder->dataArray().constVectorQint32().count() != ){
+//		qDebug() << "FATAL ERROR, MISMATCH BETWEEN RECEIVED DATA SIZE OF SCALER AND REQUESTED SIZE. RECEIVED: " << asScalarDataHolder->dataArray().constVectorQint32().count() << "REQUESTED: " << scalerChannelVectors.count();
+//		return;
+//	}
+
+	for(int x = 0; x < totalCount; x++){
+		AMDSLightWeightScalarDataHolder *asScalarDataHolder = qobject_cast<AMDSLightWeightScalarDataHolder*>(scalerClientDataRequst->data().at(x));
+		if(asScalarDataHolder){
+			QVector<qint32> oneVector = asScalarDataHolder->dataArray().constVectorQint32();
+
+			int tempRunningSum;
+			QString channelString;
+			for(int y = 0, ySize = oneVector.count(); y < ySize; y++){
+				channelString = scalerChannelIndexMap.value(y);
+				tempRunningSum = scalerChannelRunningSums.value(channelString);
+				tempRunningSum += oneVector.at(y);
+				scalerChannelRunningSums[channelString] = tempRunningSum;
+
+				if( (((x+1)*baseScalerTimeScale) % largestBaseTimeScale) == 0){
```

Comment:
I'm not 100% sure what this is doing.  I understand that we need to rebase it, but I think I'm missing a key detail though.
In any case, this conditional does not depend on `y` and should therefore not be inside the inner loop.